updating/updating-restricted-network-cluster/mirroring-image-repository.adoc

```diff
@@ -0,0 +1,171 @@
+:_content-type: ASSEMBLY
+[id="mirroring-ocp-image-repository"]
+= Mirroring the OpenShift Container Platform image repository
+include::_attributes/common-attributes.adoc[]
+:context: mirroring-ocp-image-repository
+
+toc::[]
+
+You must mirror container images onto a mirror registry before you can update a cluster in a disconnected environment. You can also use this procedure in connected environments to ensure your clusters only use container images that have satisfied your organizational controls on external content.
+
+[NOTE]
+====
+This registry must be running at all times as long as the cluster is running.
+====
+
+There are two supported methods for mirroring images onto a mirror registry:
+
+* Using the oc-mirror OpenShift CLI (`oc`) plugin
+
+* Using the `oc adm release mirror` command
+
+Compared to using `oc adm release mirror`, the oc-mirror plugin has the following advantages:
+
+* It is simpler to use.
+
+* It can mirror content other than container images.
+
+* After mirroring images for the first time, it is easier to update images in the registry.
+
+Choose one of the supported options.
+
+[id="prerequisites_updating-mirroring-disconnected"]
+== Prerequisites
+
+* You must have a container image registry that supports link:https://docs.docker.com/registry/spec/manifest-v2-2[Docker v2-2] in the location that will host the {product-title} cluster, such as Red Hat Quay.
++
+[NOTE]
+====
+If you use Red Hat Quay, you must use version 3.6 or later with the oc-mirror plugin. If you have an entitlement to Red Hat Quay, see the documentation on deploying Red Hat Quay link:https://access.redhat.com/documentation/en-us/red_hat_quay/3.6/html/deploy_red_hat_quay_for_proof-of-concept_non-production_purposes/[for proof-of-concept purposes] or link:https://access.redhat.com/documentation/en-us/red_hat_quay/3.6/html/deploy_red_hat_quay_on_openshift_with_the_quay_operator/[by using the Quay Operator]. If you need additional assistance selecting and installing a registry, contact your sales representative or Red Hat Support.
+====
++
+If you do not already have an existing solution for a container image registry, subscribers of {product-title} are provided a xref:../../installing/disconnected_install/installing-mirroring-creating-registry.adoc#installing-mirroring-creating-registry[mirror registry for Red Hat OpenShift]. The _mirror registry for Red Hat OpenShift_ is included with your subscription and is a small-scale container registry that can be used to mirror the required container images of {product-title} in disconnected installations and updates.
+
+[id="updating-restricted-network-mirror-host"]
+== Preparing your mirror host
+
+Before you perform the mirror procedure, you must prepare the host to retrieve content and push it to the remote location.
+
+include::modules/cli-installing-cli.adoc[leveloffset=+2]
+
+// Installing the oc-mirror OpenShift CLI plugin
+include::modules/oc-mirror-installing-plugin.adoc[leveloffset=+2]
```

Comment:
Isn't this part only necessary if you're using the oc-mirror method? Right now, the way it's laid out in this assembly, it looks like something you'd do no matter which method you're using.

Reply:
Regarding your question, I think that's right. I was actually hoping to receive feedback/suggestions for how to make it clearer to users that this is only necessary if you're going to use oc-mirror. From the list of different approaches below, do any seem viable/preferable? Or do you have alternative suggestions?

1. Adding a note above the first paragraph in the module (wrapped in an ifdef so it only shows up on this assembly) that says something to the effect of "this procedure is only necessary if you are going to use oc-mirror to mirror your images."
2. Creating a heading titled something like "additional preparation for using oc-mirror", nesting the heading within the "preparing your mirror host" section, and then putting this "Installing oc-mirror" module within the "additional prep" subsection.
3. Giving up on trying to fit this module in the "preparing your mirror host" section altogether, and just finding somewhere to place it in the "Mirroring resources using the oc-mirror plugin" section of this assembly